calicoctl/calico_ctl/status.py

```diff
@@ -99,6 +100,57 @@ def status(arguments):
         print "No IPv6 address configured.\n"
 
 
+def check_container_status_docker():
+    """
+    Checks and prints the calico/node container status when running in Docker. 
+    """
+    try:
+        calico_node_info = filter(lambda container: "/calico-node" in
+                                  container["Names"],
+                                  docker_client.containers())
+        if len(calico_node_info) == 0:
+            print "calico-node container not running"
+            sys.exit(1)
+        else:
+            print "calico-node container is running. Status: %s" % \
+                  calico_node_info[0]["Status"]
+
+            libraries_cmd = docker_client.exec_create("calico-node",
+                                                      ["sh", "-c",
+                                                       "cat libraries.txt"])
+            libraries_out = docker_client.exec_start(libraries_cmd)
+            result = re.search(r"^calico\s*\((.*)\)\s*$", libraries_out,
+                               re.MULTILINE)
+
+            if result is not None:
+                print "Running felix version %s" % result.group(1)
+    except ConnectionError:
+        print "Docker is not running"
+        # TODO: Perform status checks in platform-independent way.
+        sys.exit(1)
+
+
+def check_container_status_rkt():
+    """
+    Checks and prints the calico/node container status when running in rkt. 
+    """
+    list_cmd = ["sudo", "rkt", "list"]
+    p = Popen(list_cmd, stdin=PIPE, stdout=PIPE, stderr=PIPE)
```

Comment:
It doesn't capture stderr should something go wrong.  This let's us handle the error better.